lib/config/expression.cpp

```diff
@@ -37,6 +38,10 @@ void Expression::ScriptBreakpoint(ScriptFrame& frame, ScriptError *ex, const Deb
 
 ExpressionResult Expression::Evaluate(ScriptFrame& frame, DebugHint *dhint) const
 {
+	Defer decreaseStackDepth([&frame]{
+		frame.DecreaseStackDepth();
+	});
```

Comment:
How is this supposed to work, when an exception gets raised within the `IncreaseStackDepth()` function?

Reply:
It won't be executed then. But that's fine: if `IncreaseStackDepth()` throws, then it did not actually increment anything, so there's nothing to undo:

https://github.com/Icinga/icinga2/blob/3a8abdcc3b85a3be62a0805b465e5fac55ae0538/lib/base/scriptframe.cpp#L82-L88

Reply:
> It won't be executed then. But that's fine

Are you sure about that? After all, previously `DecreaseStackDepth()` was always being called, regardless of whether `IncreaseStackDepth()` raised an exception or not.

Reply:
`frame.Depth` should have the same value before and after calling `Expression::Evaluate()`. That's not the case with the current implementation and this would be fixed by this change, so maybe this also qualifies for a separate bugfix PR.